lib/apparmortest.pm

```diff
@@ -404,17 +404,21 @@ sub adminer_setup {
     assert_script_run("mv $adminer_file $adminer_dir");
 
     # Test Adminer can work
+    select_console 'x11';
     x11_start_program("firefox http://localhost/adminer/$adminer_file", target_match => "adminer-login", match_timeout => 300);
 
     # Exit x11 and turn to console
-    send_key_until_needlematch("generic-desktop", 'alt-f4', 2, 5);
+    send_key "alt-f4";
+    # Send "ret" key in case of any pop up message
+    send_key_until_needlematch("generic-desktop", 'ret', 5, 5);
```

Comment:
isn't alt-f4 suppose to close popups the same way as ret? problem is that by hitting ret you can agree with something that you don't want to , while alt-f4 is just closing. 
another thing that in `adminer_database_delete` we having basically same flow ( open browser with url / check needle / close browser ) but the way browser is closed is different . 
In case we are in mode "I don't care what app is saying to me I just need to get rid from it" why not just kill it via killall in terminal ?

Reply:
Thank you for your review.
1.
On Tumbleweed run if test case opens the firefox there  will be one more new tab opened (the title is "Firefox Privacy Notice" ), then if I want to close the firefox a 'Cofirm close: "Close tabs" ' message window will pop up, the "alt-f4" can close the pop-up window, but can not close the firefox by executing "Close tabs". if only using "alt-f4" keys the closing of "firefox" will fall into endless loop, instead the key "ret" can execute "Close tabs".

More, in the process I know what I am doing so the "something that you don't want to" seems/probably will not happen.

2. 
Yes, you are right. there are many ways to kill/close the firefox, 
In this test case I have to change from "x11" to "text console terminal" to run "# kill ...".
And using "alt-f4" to kill makes the sub function more understandable and clear. 

I would like to keep the code.
Thanks for your time and ideas.